### ISSUE TITLE
broken package procedure name when procedure is called

### ISSUE THREAD

**Issue by okbob** [darold/ora2pg#503]
We have a package:
<pre>
CREATE OR REPLACE PACKAGE IDES_JMMAJ.RRRUTIL AS
   PROCEDURE R_G_PROVEST_PREPOCET_SET( mNASTAV IN CHAR );
 FUNCTION R_G_PROVEST_PREPOCET   RETURN  CHAR;
END;
</pre>
This procedure is called from trigger:
<pre>
TRIGGER NAJPOLOZ_AU
  after update on najpoloz
  for each row

declare
  -- local variables here
  --A CHAR(1);
  mRecNajpoloz NAJPOLOZ%ROWTYPE;
  mZmenaPodpol boolean;
  mID_NAJPROS NAJPROS.ID_NAJPROS%TYPE;
  --mChyba boolean;
  mKOD_CTYPPRED_POM CTYPPRED.KOD_CTYPPRED%TYPE;
  mCVZNIKRADK_POM  NAJPREDP.CVZNIKRADK%TYPE;
  mZP_VYT NAJVYH07.ZP_VYT%TYPE;
  --mPOCET_RENE NUMBER;
  mNajpoloz_rec_pom NAJPOLOZ%ROWTYPE;
begin

   -- REN PRIDANO 19.08.2009 kvuvi zmene typu vypoctu pouze u NV
   --- meni se pouze pokud je bezny predpis.
   mKOD_CTYPPRED_POM := 'B';
   mCVZNIKRADK_POM := RRRUTIL.R_IMPORT_DES();
   BEGIN
     SELECT KOD_CTYPPRED , CVZNIKRADK INTO mKOD_CTYPPRED_POM , mCVZNIKRADK_POM
       FROM V_NAJPREDP_BEZLOKCAS
       WHERE ID_PREDPIS = :new.ID_PREDPIS;
   EXCEPTION
     WHEN OTHERS THEN
       mKOD_CTYPPRED_POM := 'B';
       mCVZNIKRADK_POM := RRRUTIL.R_IMPORT_DES();
   END;
    /*
    UPDATE NAJPREDP
      SET ECASTKACELKEM = ( ECASTKACELKEM - :old.EPOL_CASTKA +  :new.EPOL_CASTKA )
      WHERE ID_PREDPIS = :old.ID_PREDPIS ;

   */

   mRecNajpoloz.ID_DPLA := :new.ID_DPLA;
   mRecNajpoloz.ID_POL_PRED := :new.ID_POL_PRED;
   mRecNajpoloz.Id_Cnajvyppol := :new.ID_CNAJVYPPOL;
   mRecNajpoloz.Id_Polonaj := :new.ID_POLONAJ;
   mRecNajpoloz.Id_Najvyh176 := :new.ID_NAJVYH176;
   mRecNajpoloz.Najpoc := :new.NAJPOC;
   mRecNajpoloz.Id_Najpros := :new.ID_NAJPROS;
   mRecNajpoloz.Jed_Castka := :new.JED_CASTKA;
   mRecNajpoloz.Epol_Castka := :new.EPOL_CASTKA;
   mRecNajpoloz.Platnost_Od := :new.PLATNOST_OD;
   mRecNajpoloz.Id_Predpis := :new.ID_PREDPIS;
   mRecNajpoloz.KOD_CNAJVYPPOL := :new.KOD_CNAJVYPPOL;
   mRecNajpoloz.ZIV := :new.ZIV;
   mRecNajpoloz.NAJPOC := :new.NAJPOC;
   mRecNajpoloz.ID_ASPRAVCE := :new.ID_ASPRAVCE;
   mRecNajpoloz.ID_APROVOZNA := :new.ID_APROVOZNA;
   mRecNajpoloz.LIDENT := :new.LIDENT;
   mRecNajpoloz.ID_VZOR := :new.ID_VZOR;
   mRecNajpoloz.OBDOBI_DO := :new.OBDOBI_DO;
   mRecNajpoloz.OBDOBI_OD := :new.OBDOBI_OD;
   mRecNajpoloz.POC_OSO := :new.POC_OSO;
   mRecNajpoloz.SAZBA_POL := :new.SAZBA_POL;
   mRecNajpoloz.DPH_DOP := :new.DPH_DOP;
   mRecNajpoloz.DPH_PROC := :new.DPH_PROC;
   mRecNajpoloz.ZAOKR_ZAKLAD := :new.ZAOKR_ZAKLAD;
   mRecNajpoloz.ZAOKR_CELKEM := :new.ZAOKR_CELKEM;

   ---IF :new.KRAT_MES <> :old.KRAT_MES THEN
   mRecNajpoloz.Krat_Mes := :new.KRAT_MES ;
   ---END IF ;
  mZmenaPodpol := False;

  IF (( TRIM(:new.KOD_CNAJVYPPOL) = 'VYPO' )
       OR ( TRIM(:new.KOD_CNAJVYPPOL ) IN ( 'A' , 'B' ,'C', 'D' , 'G' ,'S' ,'F','K','H'  )) )THEN

        IF ( TRIM(:new.KOD_CNAJVYPPOL) = 'VYPO') AND
          ( :old.ID_CNAJVYPPOL <> :new.ID_CNAJVYPPOL ) THEN
           mZmenaPodpol := True;
        END IF;
        /*
        INSERT INTO RRRNAJUZA_ERR ( AKCE , TEXT , CHYBA )
          VALUES ( 'najpoloz_au_1', :new.ZMENA_VYP , NULL );
        */
        IF ( (TRIM(mRecNajpoloz.KOD_CNAJVYPPOL ) IN (  'D'  ,'S' , 'F' , 'C' ,'K','H' ))
            AND (( :old.ID_CNAJVYPPOL <> :new.ID_CNAJVYPPOL )
                --OR ( :new.JED_CASTKA <> :old.JED_CASTKA ) OR ( RRRUTIL.G_PROVEST_PREPOCET = 'A'  ) ))  THEN
                OR ( :new.JED_CASTKA <> :old.JED_CASTKA ) OR ( RRRUTIL.R_G_PROVEST_PREPOCET() = 'A' ) ))  THEN
              /*
              INSERT INTO RRRNAJUZA_ERR ( AKCE , TEXT , CHYBA )
                   VALUES ( 'najpoloz_au_2', null , NULL );
              */
                mID_NAJPROS := :new.ID_NAJPROS;
                ---mID_NAJPROS := :new.JED_CASTKA;
                IF (( TRIM(mKOD_CTYPPRED_POM) = 'B' ) AND  ( mCVZNIKRADK_POM <> RRRUTIL.R_IMPORT_DES() )) THEN
                  UPDATE NAJPROS
                   SET SAZBA = :new.JED_CASTKA,
                       TYP_NAJ = TRIM(mRecNajpoloz.KOD_CNAJVYPPOL )
                   WHERE ID_NAJPROS = :new.ID_NAJPROS;
                 END IF;
                 mZmenaPodpol := True;

         END IF;
         IF ((TRIM(mRecNajpoloz.KOD_CNAJVYPPOL ) IN (  'A' , 'B' , 'G'  ))
              AND (( :old.ID_CNAJVYPPOL <> :new.ID_CNAJVYPPOL ) OR ( RRRUTIL.R_G_PROVEST_PREPOCET() = 'A'  ) ) ) THEN
                  mZmenaPodpol := True;
                IF TRIM(mKOD_CTYPPRED_POM) = 'B' THEN
                  UPDATE NAJPROS
                    SET   TYP_NAJ = TRIM(mRecNajpoloz.KOD_CNAJVYPPOL )
                   WHERE ID_NAJPROS = :new.ID_NAJPROS;
                END IF;

         END IF;
         --- REN 20.05.2010 - aktualizace sazby do najpros.
         IF (( TRIM(:new.KOD_CNAJVYPPOL ) =  'C' ) AND ( :new.ID_NAJVYH07 IS NOT NULL )) THEN
            BEGIN
              SELECT ZP_VYT INTO mZP_VYT
                FROM NAJVYH07
                WHERE ID_NAJVYH07 = :new.ID_NAJVYH07;

               IF mZP_VYT = 'J' THEN
                  UPDATE NAJPROS
                    SET SAZBA = :new.SAZBA_POL
                    WHERE ID_NAJPROS = :new.ID_NAJPROS;

               END IF;
            EXCEPTION
              WHEN OTHERS THEN
              mZP_VYT := NULL;
            END;
         END IF;

        /*
        RRRUTIL.G_HODNOTY_NAJPOLOZ.ID_POL_PRED := :new.ID_POL_PRED;
        RRRUTIL.G_HODNOTY_NAJPOLOZ.ID_POLONAJ := :new.ID_POLONAJ;
        RRRUTIL.G_HODNOTY_NAJPOLOZ.ID_NAJPROS := :new.ID_NAJPROS;
        */
        
        mNajpoloz_rec_pom.Id_Pol_Pred := :new.ID_POL_PRED;
        mNajpoloz_rec_pom.ID_POLONAJ := :new.ID_POLONAJ;
        mNajpoloz_rec_pom.ID_NAJPROS := :new.ID_NAJPROS;
        RRRUTIL.R_g_hodnoty_najpoloz_set(mNajpoloz_rec_pom );

        IF mZmenaPodpol = True THEN
           INSERT INTO RRRNAJUZA_ERR ( AKCE , TEXT , CHYBA )
                   VALUES ( 'najpoloz_au_2', null , NULL );

           mRecNajpoloz.GEN_PREDPIS := 'P'; --- kdyz je update tak pouze  z Parametru
           DELETE FROM NAJPODPOL WHERE ID_POL_PRED = :new.ID_POL_PRED AND GEN_APP = 'A' ;
           TVORBA_PODPOLOZEK( mRecNajpoloz );
           --raise_application_error( -20010 , 'ddddddddddd' );
        ELSE
          IF ( TRIM(:new.KOD_CNAJVYPPOL) = 'VYPO' )
            OR ( TRIM(:new.KOD_CNAJVYPPOL ) IN ( 'A' ,'B' ,'C', 'D' , 'G' , 'S' , 'F','K','H' )) THEN
             IF :new.KRAT_MES <> :old.KRAT_MES THEN
                UPDATE NAJPODPOL
                  SET KRAT_MES = :new.KRAT_MES
                  WHERE ID_POL_PRED = :new.ID_POL_PRED  ;
           END IF ;
        END IF ;

     END IF;

  END IF ;

  IF :new.POC_OSO <> :old.POC_OSO THEN

      UPDATE NAJPODPOL
        SET POC_OSO = :new.POC_OSO
        WHERE ID_POL_PRED = :new.ID_POL_PRED;

     INSERT INTO RRRNAJUZA_ERR( AKCE )
           VALUES ( 'PODPOL_OSO' );

   END IF ;

  IF :new.OBDOBI_DO <> :old.OBDOBI_DO THEN

      UPDATE NAJPODPOL
        SET OBDOBI_DO = :new.OBDOBI_DO
        WHERE ID_POL_PRED = :new.ID_POL_PRED ;

  END IF ;

  IF :new.ID_DPLA  <> :old.ID_DPLA THEN
    NAJPOLOZ_SPOJCISLO( mRecNajpoloz  );
  END IF;

  IF ( TRIM(:new.KOD_CNAJVYPPOL) = 'ZP' ) THEN
      UPDATE NAJPOLZRP
        SET ID_POL_PRED = ID_POL_PRED_POM
        WHERE ID_POL_PRED_POM = mRecNajpoloz.ID_POL_PRED
          AND ID_POL_PRED IS NULL ;
  END IF;

  ---RRRUTIL.G_PROVEST_PREPOCET := 'B';
  RRRUTIL.R_G_PROVEST_PREPOCET_SET('B');

  --- REN 31.03.2011 POKUD DOJDE KE ZMENE CASTY ZAPSAT DO LOGU
  --- NASTAVENO POUZE U AVALU
  /*
  IF :old.EPOL_CASTKA <> :new.EPOL_CASTKA THEN
     INSERT INTO RRRNAJUZA_ERR ( AKCE , INFO , CHYBA )
      VALUES ( 'EPOL_CASTKA', :new.ID_POL_PRED ,:old.EPOL_CASTKA || '----' || :new.EPOL_CASTKA || '----' || :new.LIDENT  );

  END IF;
  */

end NAJPOLOZ_AU;
</pre>
Unfortunately `RRRUTIL.R_G_PROVEST_PREPOCET_SET('B');` is translated badly `PERFORM rrrutil.r_g_provest_prepocet()_set('B');` - should be `PERFORM rrrutil.r_g_provest_prepocet_set('B');`

Postgres:
<pre>
  IF ( btrim(NEW.KOD_CNAJVYPPOL) = 'ZP' ) THEN
      UPDATE NAJPOLZRP
        SET ID_POL_PRED = ID_POL_PRED_POM
        WHERE ID_POL_PRED_POM = mRecNajpoloz.ID_POL_PRED
          AND ID_POL_PRED = NULL;
  END IF;
              
              
  ---RRRUTIL.G_PROVEST_PREPOCET := 'B';
  PERFORM rrrutil.r_g_provest_prepocet()_set('B');
</pre>

**Comment by darold** [comment]
I can not test it right now but perhaps you can confirm that last commit ac55ab1 solves the issue?

**Comment by okbob** [comment]
yes, it is fixed - Thank you. It is important milestone for us - All triggers are imported to PostgreSQL without error detected by plpgsql_check.

**Comment by darold** [comment]
Very good news Pavel. I have used plpgsql_check for a migration next month and it's a very helpful tool, especially for trigger that can not be tested easily. Congratulation for this tool.

I have done some changes in the outer join and connect by translation. Translation of connect by was just broken in complex queries and all right outer join are now translated into left outer join to simplify the code. I've also fixed the filters of the outer join, filter with a constant was wrongly reported to the where clause. I hope it will not break something in your actual migration works. Thanks again for all feedbacks.
